### ISSUE TITLE
Are customized attachment  folder names possible (answer == yes)

### ISSUE THREAD

**Issue by vguogis** [thundernest/import-export-tools-ng#336]
Hello,

It could be great to have an opportunity to customize folder names also. Here is my simple attempt to save attachments for plain text export at folders named as "Extended file name". Sorry, I'm not familiar with JS and programming for TB so just copied and changed part of code.

```
// vg-added
			var sub1;
			if (!hdrArray)
				sub1 = getSubjectForHdr(hdr, file.path);
			else {
				var parts = hdrArray[IETexported].split("§][§^^§");
				sub1 = parts[4];
				sub1 = sub.replace(/[\x00-\x1F]/g, "_");
			}

			sub1 = IETstr_converter(sub1);



							// attachmentsExtendedFilenameFormat = "1";
							if (attachmentsExtendedFilenameFormat === "") {
								// vg-changed
								// attDirContainer.append("Attachment");
								attDirContainer.append(sub1);
							} else {
								let afname = constructAttachmentsFilename(1, hdr);
								attDirContainer.append(afname);

							}
```

**Comment by cleidigh** [comment]
@vguogis 
This is already supported for both attachment folder names and embedded image attachments.
The file name format follows the regular formatting except you have to configure in the Thunderbird config editor  accessed from the bottom of the general preferences.  There you can set 
attachments.filename_extended_format
embedded_attachments.filename_extended_format

my settings shown in the screenshot is ${subject}-${date_custom}-${index}
i have not figured how I want this in the ui yet.
Let me know if I have understood your goals and if you have any questions.
@cleidigh

**Comment by vguogis** [comment]
Hi @cleidigh 
Thank you very much! This is exactly what I was looking for.
Now I'm experimenting with	_[${date_custom} -${sender_email} +${recipient_email} =${subject}]_ format for a better file search by from/to/subject. At least it will be easy to use multirename if required.

**Comment by cleidigh** [comment]
@vguogis 
Great! I was hoping that solved your problem.
I'm limited in my development so I just have not been able to make the UI additions.  I have some other hidden things like hot keys, but they are listed in the help...
I will create an issue for adding this and you can track the progress.
@cleidigh

**Comment by vguogis** [comment]
@cleidigh 
Sometimes is good to learn about hidden featues. It's solved for me but I believe this option could be very useful for someone else too.

**Comment by cleidigh** [comment]
@vguogis 
Yes, curious, did you use the help at all? I may need a help button other than just question marks...
@cleidigh

**Comment by vguogis** [comment]
@cleidigh 
Yes, question marks for help are good for me, I have clicked to look what's inside.

**Comment by cleidigh** [comment]
@vguogis 
Well here is something for your weekend, I took some time off bug work.
As the first beta user you get to give feedback on the ui.
It's not localized yet. Greek I think? ( I'm Greek by marriage)
Im debating what I should have as defaults, original like " Attachments"
or something with the tokens as example.
Thoughts?

https://github.com/thundernest/import-export-tools-ng/blob/v12.0.2/xpi/import-export-tools-ng-12.0.2-b1-tb.xpi

@cleidigh

**Comment by vguogis** [comment]
Hi @cleidigh 
Unfortunately I can't load this file. about:addons says it's damaged.
I will take a look at localization files after. At the moment I use English version.

**Comment by cleidigh** [comment]
@vguogis 
I had another user with a corrupt download. If I recall he used a different browser, but I am not sure. I tried downloading now with Chrome and it was fine.
Don't worry about localization, in fact this gives me the excuse to add the lt locale!
Hopefully you can figure out the download issue, btw i distribute the betas this way without any problems Sometimes just glitches .
@cleidigh

**Comment by vguogis** [comment]
@cleidigh 
Good, downloaded with Chrome. The problem was with direct download from TB.

**Comment by cleidigh** [comment]
ah good 
let me know what you think 
@cleidigh

**Comment by vguogis** [comment]
@cleidigh 
Looks good. I already checked for custom folder naming fields. And I like flexibility of tokens usage - even custom date it is possible to format.

**Comment by cleidigh** [comment]
yes I spent a bunch of time on on the tokens and date features.
Great, it was time to get these exposed.

@cleidigh

**Comment by cleidigh** [comment]
Added in b1, will be in v12.0.2

**Comment by cleidigh** [comment]
Addressed with v12.0.2 Release
